Mathlib/Data/Setoid/Basic.lean

```diff
@@ -196,6 +196,15 @@ theorem eq_top_iff {s : Setoid α} : s = (⊤ : Setoid α) ↔ ∀ x y : α, s x
   rw [_root_.eq_top_iff, Setoid.le_def, Setoid.top_def]
   simp only [Pi.top_apply, Prop.top_eq_true, forall_true_left]
 
+nonrec theorem subsingleton_iff {s : Setoid α} : Subsingleton (Quotient s) ↔ s = ⊤ := by
+  simp_rw [subsingleton_iff, Quotient.forall, Quotient.eq, eq_top_iff]
```

Comment:
This already exists

https://github.com/leanprover-community/mathlib4/blob/c728e645fccf7166c3ac71ce0c1ca1a32c10a268/Mathlib/Data/Setoid/Basic.lean#L444-L449